flytekit/clis/sdk_in_container/fast_register.py

```diff
@@ -0,0 +1,164 @@
+
+        if test:
+            click.echo("Would fast register {:20} {}".format("{}:".format(o.entity_type_text), o.id.name))
+        else:
+            click.echo("Fast registering {:20} {}".format("{}:".format(o.entity_type_text), o.id.name))
+            _get_additional_distribution_loc(_sdk_config.FAST_REGISTRATION_DIR.get(), digest)
+            if o.resource_type == _identifier_model.ResourceType.TASK:
+                o.fast_register(project, domain, o.id.name, digest, remote_package_path)
+            else:
+                o.register(project, domain, o.id.name, digest)
+
+
+def fast_register_tasks_only(
+    project: str, domain: str, pkgs: _List[str], test: bool, version: str, source_dir: _os.PathLike
+):
+    if test:
+        click.echo("Test switch enabled, not doing anything...")
+
+    if not version:
+        digest = _compute_digest(source_dir)
+    else:
+        digest = version
+    remote_package_path = _upload_package(source_dir, digest, _sdk_config.FAST_REGISTRATION_DIR.get())
+
+    click.echo(
+        "Running task only fast registration for {}, {}, {} with version {} and code dir {}".format(
+            project, domain, pkgs, digest, source_dir
+        )
+    )
+
+    # Discover all tasks by loading the module
+    for m, k, t in iterate_registerable_entities_in_order(pkgs, include_entities={_task.SdkTask}):
+        name = _utils.fqdn(m.__name__, k, entity_type=t.resource_type)
+
+        if test:
+            click.echo("Would fast register task {:20} {}".format("{}:".format(t.entity_type_text), name))
+        else:
+            click.echo("Fast registering task {:20} {}".format("{}:".format(t.entity_type_text), name))
+            t.fast_register(project, domain, name, digest, remote_package_path)
+
+
+@click.group("fast-register")
+@click.option("--test", is_flag=True, help="Dry run, do not actually register with Admin")
+@click.pass_context
+def fast_register(ctx, test=None):
+    """
+    Run registration steps for the Flyte entities in this container. This is an optimization to avoid the conventional
+    container build and upload cycle. This can be useful for fast iteration when making code changes. If you do need
+    to change the container itself (e.g. by adding a new dependency/import) you must rebuild and upload a container.
+
+    Caveats: Your flyte config must specify a fast registration dir like so:
+    [sdk]
+    fast_registration_dir=s3://my-s3-bucket/dir
+
+    **and** ensure that the role specified in [auth] section of your config has read access to this remote location.
+    Furhtermore, the role you assume to call fast-register must have **write** permission to this remote location.
+
+    Run with the --test switch for a dry run to see what will be registered.  A default launch plan will also be
+    created, if a role can be found in the environment variables.
+    """
+
+    ctx.obj[CTX_TEST] = test
+    ctx.obj[CTX_CURRENT_DIR] = _os.getcwd()
```

Comment:
is this necessary?  can we get this command to follow the regular register semantics?  should either follow the --pkgs switch or the workflow_package configuration setting.

Reply:
or is this actually the goal?  like if a user cd's into project/app/workflows/blah and runs fast-register, is the goal to only pick up entries in that folder?

Reply:
It's not sufficient to follow the pks config because there's nothing to stop users from defining code outside of the specific workflow packages (but still in the repo). I made it cwd as a convenience, but perhaps it's better to explicitly require them to provide their code root dir. I'll update